### PR TITLE
Prettier

### LINKED ISSUE #31: Add prettier to enforce code formatting across IDEs

### PR DESCRIPTION
**Pull request by dsimansk**
Resolves #31, this PR adds [prettier](https://github.com/prettier/prettier) library for formatting purposes and pre-commit hook to execute format cmd before every commit. 

@jludvice pls review this, if you don't like it or want to add more rules.

@lvydra be aware that since this is changing almost all files formatting you'll need to update your fork once it's merged to prevent unnecessary conflicts.

### CODE REVIEW

File: e2e/app.po.ts [hunk 1]
```diff
@@ -217,25 +216,29 @@ export class AppPage {
       await new GithubLogin().authorizeApp();
     }
 
-    await browser.wait(ExpectedConditions.presenceOf(this.rootElement), 30 * 1000,
-    'syndesis root element - assuming we are already logged in');
+    await browser.wait(
+      ExpectedConditions.presenceOf(this.rootElement),
+      30 * 1000,
+      'syndesis root element - assuming we are already logged in'
```
Comment (jludvice) [inline e2e/app.po.ts]: Can we get around method parameters on diferent lines somehow?

File: e2e/app.po.ts [hunk 2]
```diff
@@ -217,25 +216,29 @@ export class AppPage {
...
-        callback(data);
-      });
-    }).then(jsonString => JSON.parse(<string> jsonString));
+    return browser.driver
+      .executeAsyncScript(callback => {
+        jQuery.get('/config.json', function(data) {
+          callback(data);
+        });
+      })
+      .then(jsonString => JSON.parse(<string>jsonString));
```
Comment (jludvice) [inline e2e/app.po.ts]: have you tried `yarn lint` after this reformat? I recall some complains in similar situations like this one

File: e2e/common/common.steps.ts [hunk 3]
```diff
@@ -150,37 +140,40 @@ class CommonSteps {
...
     const element = this.world.app.getElementByClassName(elementClassName);
-    expect(element.isPresent(), `There must be present a element ${elementClassName}`)
-      .to.eventually.be.true;
+    expect(element.isPresent(), `There must be present a element ${elementClassName}`).to.eventually.be.true;
 
-    expect(element.isPresent(), `There must be enabled element ${elementClassName}`)
-      .to.eventually.be.true.notify(callback);
+    expect(element.isPresent(), `There must be enabled element ${elementClassName}`).to.eventually.be.true.notify(
+      callback
+    );
```
Comment (jludvice) [inline e2e/common/common.steps.ts]: This line break feels weird :(

File: e2e/common/common.steps.ts [hunk 4]
```diff
@@ -208,13 +201,12 @@ class CommonSteps {
...
   @when(/^scroll "([^"]*)" "([^"]*)"$/)
   public async scroll(topBottom: string, leftRight: string): P<any> {
-
     const size = await browser.manage().window().getSize();
     const directions: Object = {
       top: 0,
       bottom: size.height,
       left: 0,
-      right: size.width,
+      right: size.width
```
Comment (jludvice) [inline e2e/common/common.steps.ts]: not sure what's the rule for trailing comma in `yarn lint`

File: e2e/integrations/edit/datamapper.po.ts [hunk 5]
```diff
@@ -48,18 +48,13 @@ export class DataMapperComponent implements SyndesisComponent {
...
     const columns = await this.dataMapperColumns();
     const src = columns[0];
     const dest = columns[1];
 
-
-    return P.all([
-      this.selectMapping(source, src),
-      this.selectMapping(target, dest),
-    ]);
+    return P.all([this.selectMapping(source, src), this.selectMapping(target, dest)]);
```
Comment (jludvice) [inline e2e/integrations/edit/datamapper.po.ts]: Personally I don't like the way code looks after this reformat :(
for instance here I wanted to have different function calls on different lines and there is no way around that reformat :(
```ts
    return P.all([		
      this.selectMapping(source, src),		
      this.selectMapping(target, dest),		
    ]);
```

**Comment by dsimansk** [comment]
I'll close this PR for now. If we feel like adding code-formatter in future, feel free to reopen the linked issue.

**Pull request closed without merging**